live-examples/js-examples/promise/promise-finally.js

```diff
@@ -0,0 +1,17 @@
+function hasMailArrived() {
```

Comment:
Does that name match the code as well as I could?
If turned into a question, "Has mail arrived?" you'd expect the answer to be "Yes" or "No". 
But the function returns a Promise and it has its own side-effect (the `console.log('Experiment completed');`).

It's not a great name either, but perhaps `checkMail()`? What do you think?

Reply:
I was thinking the same thing, function name could match better for what it is actually returning, which is `Promise<any>` but is expected to be `Promise<String>`. After turning it into a question, it becomes more clear the name could match the code better. 

> It's not a great name either, but perhaps checkMail()? What do you think?

Agreed, name could be better overall. I think `checkMail()` has a nice ring too it.